src/backend/src/main/java/com/linkedpipes/lpa/backend/ExecutionController.java

```diff
@@ -0,0 +1,45 @@
+package com.linkedpipes.lpa.backend;
+
+import com.linkedpipes.lpa.backend.services.HttpUrlConnector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.io.IOException;
+
+@RestController
+public class ExecutionController {
+
+    private static final Logger logger =
+            LoggerFactory.getLogger(DiscoveryController.class);
+
+    private HttpUrlConnector httpUrlConnector = new HttpUrlConnector();
+
+    @RequestMapping("/execution/status")
+    public ResponseEntity<String> getStatus(@RequestParam( value="executionIri") String executionIri) throws IOException{
+        if(executionIri == null || executionIri.isEmpty()) {
+            return new ResponseEntity("Execution IRI not provided.", HttpStatus.BAD_REQUEST);
+        }
```

Comment:
Again, having EtlServiceComponent/EtlComponent/EtlService class with hidden logic would make this much cleaner. You can already see that DiscoveryComponent and EtlComponent would share some logic for calling HTTP.

With SRP, you will achieve code that reads as simple steps and you don't care how those are implemented. And if you do, you will open implementation.